### PR TITLE
Sending own arguments to ffprobe

### PR DESCRIPTION
**Pull request by lkischler**

### CODE REVIEW

File: Xabe.FFmpeg/FFprobeWrapper.cs [hunk 1]
```diff
@@ -76,6 +81,18 @@ private int GetGcd(int width, int height)
             return width == 0 ? height : width;
         }
 
+        public async Task<T> Start<T>(string args)
+        {
+            args = "-print_format json " + args;
```
Comment (tomaszzmuda) [inline Xabe.FFmpeg/FFprobeWrapper.cs]: What if in args will be another -print_format?
And what if user don't want this serialized

File: Xabe.FFmpeg/Probe/IProbe.cs [hunk 2]
```diff
@@ -0,0 +1,25 @@
+﻿using System.Threading.Tasks;
+
+namespace Xabe.FFmpeg
+{
+    /// <summary>
+    ///     Allows to prepare and start IProbe.
+    /// </summary>
+    public interface IProbe
```
Comment (tomaszzmuda) [inline Xabe.FFmpeg/Probe/IProbe.cs]: change this to FFprobe

File: Xabe.FFmpeg/Probe/Probe.cs [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+{
+    /// <inheritdoc />
+    public class Probe : IProbe
+    {
+        /// <summary>
+        ///     Get new instance of Conversion
+        /// </summary>
+        /// <returns>IProbe object</returns>
+        public static IProbe New()
+        {
```
Comment (tomaszzmuda) [inline Xabe.FFmpeg/Probe/Probe.cs]: Why it cannot  be instantiated by constructor?

Comment (lkischler) [reply Xabe.FFmpeg/Probe/Probe.cs]: To be compatible with Conversion.New() ;) Constructor is not private here so can be instantiated by constructor.

File: Xabe.FFmpeg/Probe/IProbe.cs (hunk 2 above)
Comment (lkischler) [reply Xabe.FFmpeg/Probe/IProbe.cs]: Class for conversion is not FFConversion so I think Probe is better name

File: Xabe.FFmpeg/FFprobeWrapper.cs (hunk 1 above)
Comment (lkischler) [reply Xabe.FFmpeg/FFprobeWrapper.cs]: If don't want deserialization should use raw result form of this function. Always is getting last "-print_format" parameter - if somone choose this option and pass another format deserialization will break. This function is only helper for typical usage.

File: Xabe.FFmpeg/Probe/IProbe.cs [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+    {
+        /// <summary>
+        /// Start probe with result from console
+        /// </summary>
+        /// <param name="args">Args to pass to FFprobe</param>
+        /// <returns>Console output output</returns>
```
Comment (tomaszzmuda) [inline Xabe.FFmpeg/Probe/IProbe.cs]: "Output from console in specified format"

File: Xabe.FFmpeg.Test/ProbeTests.cs [hunk 5]
```diff
@@ -0,0 +1,22 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace Xabe.FFmpeg.Test
+{
+    public class ProbeTests
```
Comment (tomaszzmuda) [inline Xabe.FFmpeg.Test/ProbeTests.cs]: Please add test to catch FFprobeException

File: Xabe.FFmpeg/Probe/IProbe.cs (hunk 4 above)
Comment (lkischler) [reply Xabe.FFmpeg/Probe/IProbe.cs]: done

File: Xabe.FFmpeg.Test/ProbeTests.cs (hunk 5 above)
Comment (lkischler) [reply Xabe.FFmpeg.Test/ProbeTests.cs]: outdated

**Pull request merged**